thriftbp/client_pool_test.go

```diff
@@ -79,6 +80,116 @@ func TestCustomClientPool(t *testing.T) {
 	}
 }
 
+func TestBehaviorWithNetworkIssues(t *testing.T) {
+	ln, err := net.Listen("tcp", addr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ln.Close()
+
+	cfg := thriftbp.ClientPoolConfig{
+		ServiceSlug:     "test",
+		EdgeContextImpl: ecinterface.Mock(),
+		ConnectTimeout:  time.Millisecond * 5,
+		SocketTimeout:   time.Millisecond * 15,
+
+		RequiredInitialConnections: 2,
+		InitialConnections:         5,
+		MaxConnections:             10,
+	}
+
+	for _, c := range []struct {
+		label    string
+		addrGen  thriftbp.AddressGenerator
+		validate func(thriftbp.ClientPool, error)
+	}{
+		{
+			label: "network-fails-once-before-required",
+			addrGen: func() thriftbp.AddressGenerator {
+				i := 0
+				return func() (string, error) {
+					i += 1
+					var err error
+					if i == 1 {
+						err = fmt.Errorf("something broke")
+					}
+					return ln.Addr().String(), err
+				}
+			}(),
```

Comment:
🔕 Is there any other way for us to fake these errors other than hacking them into the AddressGenerator? This isn't _really_ the kind of error we would expect to get since most of these just return a static string (we added it to be compatible with some old ads code).

Not a blocker, but it might be nice to have the errors be closer to errors that we would actually expect/not rely on something that is really just a backwards compatibility hack.

Reply:
I don't know, that's how every other test I found does it so I used the same approach